summingbird-storm/src/main/scala/com/twitter/summingbird/storm/Edges.scala

```diff
@@ -113,4 +139,20 @@ private object EdgeInjections {
       (input.get(0).asInstanceOf[T1], input.get(1).asInstanceOf[T2], input.get(2).asInstanceOf[T3])
     }
   }
+
+  case class KeyValueInjection[K, V]() extends Injection[Item[(K, V)], JList[AnyRef]] {
+    override def apply(tuple: Item[(K, V)]): JList[AnyRef] = tuple match {
+      case (timestamp, (key, value)) =>
+        val list = new JAList[AnyRef](3)
+        list.add(timestamp.asInstanceOf[AnyRef])
+        list.add(key.asInstanceOf[AnyRef])
+        list.add(value.asInstanceOf[AnyRef])
+        list
+    }
+
+    override def invert(valueIn: JList[AnyRef]): Try[Item[(K, V)]] = Inversion.attempt(valueIn) { input =>
+      val (timestamp, key, value) = (input.get(0), input.get(1), input.get(2))
```

Comment:
I'm not sure scalac will optimize this out, so there could be performance penalty. May be just split to multiple lines. 
val timestamp = input.get(0).asInstanceOf[Timestamp]
...